### PR TITLE
Extract FakerRepeatabilityIntegrationTest to integration test phase

### PR DESCRIPTION
**Pull request by snuyanzin**
This should allow to avoid impact of other tests on it and should fix 
such flaky issue https://github.com/datafaker-net/datafaker/actions/runs/5596562572/jobs/10234110845#step:4:1039

### CODE REVIEW

File: src/test/java/net/datafaker/FakerTest.java [hunk 1]
```diff
@@ -313,7 +313,7 @@ void differentLocalesTest() {
     }
 
     @Test
-    @Timeout(value = 1, unit = TimeUnit.MINUTES)
+    @Timeout(value = 10, unit = TimeUnit.MINUTES)
```
Comment (kingthorin) [inline src/test/java/net/datafaker/FakerTest.java:316]: I’d suggest 2, the failures I’ve seen have been decimals between 60 and 61 sec.

Comment (snuyanzin) [reply src/test/java/net/datafaker/FakerTest.java:316]: there was one failure with 160 sec (and it exceeds 2 mins)
https://github.com/datafaker-net/datafaker/actions/runs/5597303576/jobs/10235552889#step:4:939

that's why i increased it

Comment (kingthorin) [reply src/test/java/net/datafaker/FakerTest.java:316]: Of course just to prove me wrong one just hit 65.25sec…..but 2m should still be fine.

Comment (snuyanzin) [reply src/test/java/net/datafaker/FakerTest.java:316]: it seems for macos it takes longer (no idea why), as i mentioned there was a case where it took almost 3min (so far just once)....
we can set 3...
anyway it is just a timeout which means the max duration of the task which usually ends faster

Comment (kingthorin) [reply src/test/java/net/datafaker/FakerTest.java:316]: Yeah, I was just concerned that with 10 someone might misinterpret in the future. I’m probably overthinking it.

Comment (bodiam) [reply src/test/java/net/datafaker/FakerTest.java:316]: I don't think you are overthinking, I was confused by this too, I thought the 1M generations took 9-10 minutes or so.

Comment (snuyanzin) [reply src/test/java/net/datafaker/FakerTest.java]: i changed it to 3 min, let's see if macos will be happy...

btw the test is NOT about 1M, it's about 10M =)

**Review by kingthorin — APPROVED** [review]
LGTM, thanks!

File: src/test/java/net/datafaker/FakerTest.java (hunk 1 above)
Comment (bodiam) [reply src/test/java/net/datafaker/FakerTest.java]: 10M!! That's ambitious :)

**Pull request merged**